portality/templates/doaj/article.html

```diff
@@ -208,9 +212,18 @@ <h1>&lt; Article title missing &gt;</h1>
         <p>
             <strong>
                 Abstract
-                {% for url in bibjson.get_urls("fulltext") %}
-                    | <a href="{{url}}">Full Text</a>
-                {% endfor %}
+                {% if bibjson.get_urls("fulltext") %}
+                    {% for url in bibjson.get_urls("fulltext") %}
+                        | <a href="{{url}}">Full Text</a>
+                    {% endfor %}
+                {% else %}
+                    {% if doi %}
+                        {% set url_text = normalised_doi | doi_url | safe %}
+                        {% if url_text != "" %}
+                            <a href="{{ normalised_doi | doi_url | safe }}"> Full Text</a>
```

Comment:
For a slight improvement to performance here, this should be:

```
<a href="{{ url_text }}"> Full Text</a>
```

since you created it just 2 lines above.

Reply:
ah, oversight. Sorry for that, ready in 5 minutes.